### ISSUE TITLE
data source for azurerm_api_management_subscription

### ISSUE THREAD

**Issue by CloudbptestAug** [hashicorp/terraform-provider-azurerm#20575]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Whenever an API or a resource is being used as a backend to an APIM the APIs are resources can be protected using a product. The product subscription keys need to be shared to the customers to be able to access the Onboarded APIs through APIM. 

To avoid manual share, we need to dynamically read the subscription keys from Azure to be able to store them at a strong secured place dynamically and give the secured place access to limited people to be able to get the subscription keys. 

We cannot programmatically read the subscription keys with ease as there is no data source for 'azurerm_api_management_subscription' 

We need to go for a Azure REST API request or a powershell script but both are not as easy as using a data source for terraform scripting. 

Could you provide this feature or help me if there an alternative data source for the same to read?

### New or Affected Resource(s)/Data Source(s)

new

### Potential Terraform Configuration

```hcl
data "azurerm_api_manangement_subscription" "example_susbcription"{
name ="***"
product_id ="***"
resource_group_name = "***"
api_management_name = "****"
}
```

### References

no

**Comment by CorrenSoft** [comment]
Are you creating the azurerm_api_management_subscription with Terraform? Because you could grab the keys from there and store them (unless, of course, you have any other restriction).

**Comment by teask1** [comment]
> Are you creating the azurerm_api_management_subscription with Terraform? Because you could grab the keys from there and store them (unless, of course, you have any other restriction).

Hello, maybe you know how to get master key's? (Built-in all-access subscription) I would like to put them to kv.

**Comment by CloudbptestAug** [comment]
> Are you creating the azurerm_api_management_subscription with Terraform? Because you could grab the keys from there and store them (unless, of course, you have any other restriction).

yes. but for a particular existing subscription if I need to get keys(not at time of creation) it is not possible as there is no data getting feature.

**Comment by pankajdevrani** [comment]
When is this coming?

**Comment by lightpeak** [comment]
We need this for our projects as well
